lib/trino-filesystem/src/main/java/io/trino/filesystem/Location.java

```diff
@@ -197,7 +197,7 @@ public Location parentDirectory()
             return withPath(newLocation, "");
         }
 
```

Comment:
I'd be inclined to allow calling this method for non-filename locations as well.
The required changes should probably be straightforward.
I'm not sure though whether it is needed.